components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -92,6 +93,26 @@ pub fn rtrim(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
     }))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn left(lhs: &Option<Bytes>, rhs: &Option<Int>) -> Result<Option<Bytes>> {
+    Ok(match (lhs, rhs) {
+        (Some(lhs), Some(rhs)) => {
+            if *rhs <= 0 {
+                return Ok(Some(Vec::new()));
+            }
+            let s = str::from_utf8(&*lhs).unwrap();
+            let l = *rhs as usize;
```

Comment:
I guess this might cause issues when rhs is a signed int

Reply:
Hi, I just fix conflict and the ci test just success.

Reply:
The test run on CI is randomly generated so it may not cover all scenarios. Could you take a deeper look at this and check out the affect of signed int?

Reply:
Sorry, I don't understand **the affect of signed int**.
Do you mean that when the machine is 32 bit (so the usize is u32) and rhs is i64, so `*rhs as usize` might cause issues?

Reply:
Oh, never mind. I just saw that you have checked the case `if *rhs <= 0`.